matchbook-sdk-core/src/main/java/com/matchbook/sdk/core/disruptor/handlers/UserPublisherHandler.java

```diff
@@ -0,0 +1,43 @@
+package com.matchbook.sdk.core.disruptor.handlers;
+
+import java.util.Objects;
+
+import com.lmax.disruptor.EventHandler;
+import com.lmax.disruptor.LifecycleAware;
+import com.matchbook.sdk.core.StreamObserver;
+import com.matchbook.sdk.core.disruptor.messages.UserMessage;
+import com.matchbook.sdk.core.model.dataobjects.user.Balance;
+
+public class UserPublisherHandler<T extends UserMessage> implements EventHandler<T>, LifecycleAware {
+
+    private String oldThreadName;
```

Comment:
Why do we need to keep track of the thread name and restore it when the handler stops?